### ISSUE TITLE
@wordpress/components : Add "title" attribute support to options of rendered select element

### ISSUE THREAD

**Issue by lgersman** [WordPress/gutenberg#43475]
## What problem does this address?

The SelectControl currently only supports `id`, `label`, `value` and `disabled` properties for the rendered `<option>` HTML elements. 

There is no way to specifiy the `title` attribute for the rendered `<option>` elements.

For improving UX experience it would be nice to have the `<option>` `title` attribute   addressable. 

In that case users could see a descriptive information about the hovered item.

## What is your proposed solution?

Developers should supply an additional `description` attribute within the controls `options` property which gets mapped to the rendered `<option>` HTML elements `title` attribute.   

Moreover, if an item was selected the selected item `description` (aka `title`) should be set as the rendered `<select>` HTML element `title` attribute so that hovering over the control would show up the native tooltip with the description of the currently selected item.

**Comment by afercia** [comment]
Worth reminding that WordPress has been progressively removing the title attribute from most part of the codebase. Title attributes are accessible to only a fraction of users. Keyboard users can't see them. Screen reader users may miss them, depending also on their screen reader verbosity settings.

For more details, please see https://core.trac.wordpress.org/ticket/24766
For all the work done in core to remove most of the title attributes in the admin, see https://core.trac.wordpress.org/query?keywords=~title-attribute

**Comment by andrewhayward** [comment]
Just to add to @afercia's comments here, we really shouldn't be encouraging developers to rely on the use of `title` attributes to communicate information to the user, which I'd argue calling it 'description' would probably do here. [A large subset of modern web users simply don't have the option](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns) of accessing 'descriptive information about the hovered item'.

**Comment by ciampo** [comment]
Closing as per the explanations above
